lib/maplibre-compose/src/androidMain/kotlin/dev/sargunv/maplibrecompose/core/layer/SymbolLayer.kt

```diff
@@ -33,245 +38,243 @@ internal actual class SymbolLayer actual constructor(id: String, source: Source)
   }
 
-  actual fun setSymbolAvoidEdges(avoidEdges: Expression<Boolean>) {
+  actual fun setSymbolAvoidEdges(avoidEdges: BooleanExpression) {
     impl.setProperties(PropertyFactory.symbolAvoidEdges(avoidEdges.toMLNExpression()))
   }
 
-  actual fun setSymbolSortKey(sortKey: Expression<Number>) {
+  actual fun setSymbolSortKey(sortKey: FloatExpression) {
     impl.setProperties(PropertyFactory.symbolSortKey(sortKey.toMLNExpression()))
   }
 
-  actual fun setSymbolZOrder(zOrder: Expression<SymbolZOrder>) {
+  actual fun setSymbolZOrder(zOrder: EnumExpression<SymbolZOrder>) {
     impl.setProperties(PropertyFactory.symbolZOrder(zOrder.toMLNExpression()))
   }
 
-  actual fun setIconAllowOverlap(allowOverlap: Expression<Boolean>) {
+  actual fun setIconAllowOverlap(allowOverlap: BooleanExpression) {
     impl.setProperties(PropertyFactory.iconAllowOverlap(allowOverlap.toMLNExpression()))
   }
 
-  actual fun setIconOverlap(overlap: Expression<String>) {
+  actual fun setIconOverlap(overlap: StringExpression) {
     // not implemented by MapLibre-native Android yet
     // impl.setProperties(PropertyFactory.iconOverlap(overlap.toMLNExpression()))
   }
 
-  actual fun setIconIgnorePlacement(ignorePlacement: Expression<Boolean>) {
+  actual fun setIconIgnorePlacement(ignorePlacement: BooleanExpression) {
     impl.setProperties(PropertyFactory.iconIgnorePlacement(ignorePlacement.toMLNExpression()))
   }
 
-  actual fun setIconOptional(optional: Expression<Boolean>) {
+  actual fun setIconOptional(optional: BooleanExpression) {
     impl.setProperties(PropertyFactory.iconOptional(optional.toMLNExpression()))
   }
 
-  actual fun setIconRotationAlignment(rotationAlignment: Expression<IconRotationAlignment>) {
+  actual fun setIconRotationAlignment(rotationAlignment: EnumExpression<IconRotationAlignment>) {
     impl.setProperties(PropertyFactory.iconRotationAlignment(rotationAlignment.toMLNExpression()))
   }
 
-  actual fun setIconSize(size: Expression<Number>) {
+  actual fun setIconSize(size: FloatExpression) {
     impl.setProperties(PropertyFactory.iconSize(size.toMLNExpression()))
   }
 
-  actual fun setIconTextFit(textFit: Expression<IconTextFit>) {
+  actual fun setIconTextFit(textFit: EnumExpression<IconTextFit>) {
     impl.setProperties(PropertyFactory.iconTextFit(textFit.toMLNExpression()))
   }
 
-  actual fun setIconTextFitPadding(textFitPadding: Expression<PaddingValues.Absolute>) {
+  actual fun setIconTextFitPadding(textFitPadding: PaddingExpression) {
     impl.setProperties(PropertyFactory.iconTextFitPadding(textFitPadding.toMLNExpression()))
   }
 
-  actual fun setIconImage(image: Expression<TResolvedImage>) {
+  actual fun setIconImage(image: ImageExpression) {
     impl.setProperties(PropertyFactory.iconImage(image.toMLNExpression()))
   }
 
-  actual fun setIconRotate(rotate: Expression<Number>) {
+  actual fun setIconRotate(rotate: IntExpression) {
```

Comment:
```suggestion
  actual fun setIconRotate(rotate: FloatExpression) {
```

float in native code

https://github.com/maplibre/maplibre-native/blob/72d2c70130fcd4e320cf623f9bec811108b6e201/include/mbgl/style/layers/symbol_layer.hpp#L65